### PR TITLE
Support more np.random functions in ParallelAccelerator

### PR DESCRIPTION
**Pull request by ehsantn**
This PR adds support for analysis and parallelization of many np.random functions in ParallelAccelerator.

### CODE REVIEW

File: numba/tests/test_parfors.py [hunk 1]
```diff
@@ -434,6 +436,23 @@ def test_impl(v1, v2, m1, m2):
...
 
+    @skip_unsupported
+    def test_randoms(self):
+        def test_impl(n):
+            A = np.random.standard_normal(size=(n, n))
+            B = np.random.randn(n, n)
+            C = np.random.normal(0.0, 1.0, (n, n))
+            D = np.random.chisquare(1.0, (n, n))
+            E = np.random.randint(1, high=3, size=(n, n))
+            F = np.random.triangular(1, 2, 3, (n, n))
```
Comment (stuartarchibald) [inline numba/tests/test_parfors.py]: Perhaps this ought to include calls to all the distributions being added?

File: numba/tests/test_parfors.py [hunk 2]
```diff
@@ -434,6 +436,23 @@ def test_impl(v1, v2, m1, m2):
...
+            return np.sum(A+B+C+D+E+F)
+
+        n = 128
+        # only run parallel version and don't compare results since random
+        # numbers generated in parallel are different
```
Comment (stuartarchibald) [inline numba/tests/test_parfors.py]: Whilst not ideal, could an upper bound be calculated to assert some degree of validity on the answer?

File: numba/ir_utils.py [hunk 3]
```diff
@@ -902,6 +902,54 @@ def simplify_CFG(blocks):
...
+                rhs = stmt.value
+                if (rhs.op=='getattr' and rhs.attr in arr_math
+                    and isinstance(
+                    typemap[rhs.value.name], types.npytypes.Array)):
+                    rhs = stmt.value
+                    arr = rhs.value
+                    saved_arr_arg[lhs] = arr
+                    scope = arr.scope
+                    loc = arr.loc
+                    # g_np_var = Global(numpy)
```
Comment (ehsantn) [inline numba/ir_utils.py]: These kind of comments show what is being generate in the next couple of lines, which helps readability. Do you suggestions on other ways to do this?

Comment (stuartarchibald) [inline numba/ir_utils.py]: Ah, apologies, I was reading `Global` as a function and this as dead code, this is fine :)

Comment (ehsantn) [inline numba/ir_utils.py]: It's OK. Thanks.

File: numba/parfor.py [hunk 4]
```diff
@@ -456,7 +462,17 @@ def _is_supported_npycall(self, expr):
         if expr.func.name not in self.array_analysis.numpy_calls.keys():
             return False
         call_name = self.array_analysis.numpy_calls[expr.func.name]
-        if call_name in ['zeros', 'ones', 'random.ranf']:
+        if call_name in ['zeros', 'ones', 'random.ranf', 'random.random_sample',
```
Comment (ehsantn) [inline numba/parfor.py]: Thanks for the suggestion. I refactored the code and it looks much better now.

File: numba/tests/test_parfors.py (hunk 1 above)
Comment (ehsantn) [inline numba/tests/test_parfors.py]: For readability, only one call from each class is used. The control flow is identical for calls of each class (the recent refactor shows this better). We could get the lists and generate code here though.

File: numba/tests/test_parfors.py (hunk 2 above)
Comment (ehsantn) [inline numba/tests/test_parfors.py]: I added a check with 5% tolerance (ideally we need to use the statistical properties to find a tighter bound).

**Comment by ehsantn** [comment]
Thank you for the comments @stuartarchibald . I applied them and also updated the docs. `full` and `empty` are not in the docs since they are not parallelized. Yes, `A.sum()` is supported now, so I updated the docs.

**Comment by stuartarchibald** [comment]
Thanks for the fix-ups, much appreciated. Patch looks good. Merging.

**Comment by ehsantn** [comment]
Thank you.

**Pull request merged**